### PR TITLE
Missing Mappings for Windows 10

### PR DESCRIPTION
**Pull request by ststeiger**
There are mappings missing for Windows 10.

### CODE REVIEW

**Comment by srl295** [comment]
> ststeiger seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please add the email address used for this commit to your account.

if you go to https://github.com/unicode-org/cldr/commit/89afa6f60c339af4868ed55ac5d46d8e4513847d.patch you can see the email used in the commit. Make sure it's an email that's known to github, see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user as linked above.

File: common/supplemental/windowsZones.xml [hunk 1]
```diff
@@ -246,13 +246,19 @@ For terms of use, see http://www.unicode.org/copyright.html
...
 			<mapZone other="Bahia Standard Time" territory="001" type="America/Bahia"/>
 			<mapZone other="Bahia Standard Time" territory="BR" type="America/Bahia"/>
-
-			<!-- (UTC-02:00) Coordinated Universal Time-02 -->
-			<mapZone other="UTC-02" territory="001" type="Etc/GMT+2"/>
-			<mapZone other="UTC-02" territory="BR" type="America/Noronha"/>
-			<mapZone other="UTC-02" territory="GS" type="Atlantic/South_Georgia"/>
-			<mapZone other="UTC-02" territory="ZZ" type="Etc/GMT+2"/>
-
+
```
Comment (mj1856) [inline common/supplemental/windowsZones.xml]: Whitespace is incorrect throughout this PR.  This file uses tabs, not spaces.

File: common/supplemental/windowsZones.xml [hunk 2]
```diff
@@ -246,13 +246,19 @@ For terms of use, see http://www.unicode.org/copyright.html
...
 			<mapZone other="Bahia Standard Time" territory="BR" type="America/Bahia"/>
-
-			<!-- (UTC-02:00) Coordinated Universal Time-02 -->
-			<mapZone other="UTC-02" territory="001" type="Etc/GMT+2"/>
-			<mapZone other="UTC-02" territory="BR" type="America/Noronha"/>
-			<mapZone other="UTC-02" territory="GS" type="Atlantic/South_Georgia"/>
-			<mapZone other="UTC-02" territory="ZZ" type="Etc/GMT+2"/>
-
+      
+      <!-- (UTC-02:00) Mid-Atlantic Standard Time -->
```
Comment (mj1856) [inline common/supplemental/windowsZones.xml]: The display name in comments for this zone should not have been changed.

File: common/supplemental/windowsZones.xml [hunk 3]
```diff
@@ -246,13 +246,19 @@ For terms of use, see http://www.unicode.org/copyright.html
...
+      <mapZone other="UTC-02" territory="001" type="Etc/GMT+2"/>
+      <mapZone other="UTC-02" territory="BR" type="America/Noronha"/>
+      <mapZone other="UTC-02" territory="GS" type="Atlantic/South_Georgia"/>
+      <mapZone other="UTC-02" territory="ZZ" type="Etc/GMT+2"/>
+      
+      <!-- (UTC-02:00) Mid-Atlantic Standard Time -->
```
Comment (mj1856) [inline common/supplemental/windowsZones.xml]: The `"Mid-Atlantic Standard Time"` zone has been deprecated in Windows, and thus should not appear in the CLDR.  It is marked with `IsObsolete` in the Windows registry.

Any system having reference to `"Mid-Atlantic Standard Time"` should be updated to use `"UTC-02"` instead.

File: common/supplemental/windowsZones.xml [hunk 4]
```diff
@@ -736,9 +751,13 @@ For terms of use, see http://www.unicode.org/copyright.html
...
 			<mapZone other="Central Pacific Standard Time" territory="ZZ" type="Etc/GMT-11"/>
 
-			<!-- (UTC+12:00) Anadyr, Petropavlovsk-Kamchatsky -->
-			<mapZone other="Russia Time Zone 11" territory="001" type="Asia/Kamchatka"/>
-			<mapZone other="Russia Time Zone 11" territory="RU" type="Asia/Kamchatka Asia/Anadyr"/>
+      <!-- (UTC+12:00) Anadyr, Petropavlovsk-Kamchatsky -->
+      <mapZone other="Russia Time Zone 11" territory="001" type="Asia/Anadyr"/>
+      <mapZone other="Russia Time Zone 11" territory="RU" type="Asia/Anadyr"/>
+
+      <!-- (UTC+12:00) Kamchatka -->
```
Comment (mj1856) [inline common/supplemental/windowsZones.xml]: The `"Kamchatka Standard Time"` zone has been deprecated in Windows, and thus should not appear in the CLDR.  It is marked with `IsObsolete` in the Windows registry.

Any system having reference to `"Kamchatka Standard Time"` should be updated to use `"Russia Time Zone 11"` instead.

File: common/supplemental/windowsZones.xml [hunk 5]
```diff
@@ -460,9 +466,8 @@ For terms of use, see http://www.unicode.org/copyright.html
 			<mapZone other="Belarus Standard Time" territory="001" type="Europe/Minsk"/>
 			<mapZone other="Belarus Standard Time" territory="BY" type="Europe/Minsk"/>
 
-			<!-- (UTC+03:00) Moscow, St. Petersburg, Volgograd -->
+			<!-- (UTC+03:00) Moscow, St. Petersburg -->
```
Comment (mj1856) [inline common/supplemental/windowsZones.xml]: This display name change is correct.  However, the line containing the `"RU"` territory should not have been deleted, but rather edited to remove Volgograd, as in `type="Europe/Moscow Europe/Kirov"`.

File: common/supplemental/windowsZones.xml [hunk 6]
```diff
@@ -517,7 +522,11 @@ For terms of use, see http://www.unicode.org/copyright.html
 			<mapZone other="Georgian Standard Time" territory="001" type="Asia/Tbilisi"/>
 			<mapZone other="Georgian Standard Time" territory="GE" type="Asia/Tbilisi"/>
 
-			<!-- (UTC+04:00) Yerevan -->
+      <!-- (UTC+04:00) Volgograd -->
```
Comment (mj1856) [inline common/supplemental/windowsZones.xml]: This addition is correct.

File: common/supplemental/windowsZones.xml [hunk 7]
```diff
@@ -528,15 +537,21 @@ For terms of use, see http://www.unicode.org/copyright.html
...
 			<mapZone other="West Asia Standard Time" territory="MV" type="Indian/Maldives"/>
 			<mapZone other="West Asia Standard Time" territory="TF" type="Indian/Kerguelen"/>
 			<mapZone other="West Asia Standard Time" territory="TJ" type="Asia/Dushanbe"/>
 			<mapZone other="West Asia Standard Time" territory="TM" type="Asia/Ashgabat"/>
 			<mapZone other="West Asia Standard Time" territory="UZ" type="Asia/Tashkent Asia/Samarkand"/>
 			<mapZone other="West Asia Standard Time" territory="ZZ" type="Etc/GMT-5"/>
 
-			<!-- (UTC+05:00) Ekaterinburg -->
+
+      <!-- (UTC+05:00) Kyzylorda -->
```
Comment (mj1856) [inline common/supplemental/windowsZones.xml]: This addition is correct.

File: common/supplemental/windowsZones.xml [hunk 8]
```diff
@@ -528,15 +537,21 @@ For terms of use, see http://www.unicode.org/copyright.html
 			<!-- (UTC+05:00) Ashgabat, Tashkent -->
 			<mapZone other="West Asia Standard Time" territory="001" type="Asia/Tashkent"/>
 			<mapZone other="West Asia Standard Time" territory="AQ" type="Antarctica/Mawson"/>
-			<mapZone other="West Asia Standard Time" territory="KZ" type="Asia/Oral Asia/Aqtau Asia/Aqtobe Asia/Atyrau Asia/Qyzylorda"/>
+			<mapZone other="West Asia Standard Time" territory="KZ" type="Asia/Oral Asia/Aqtau Asia/Aqtobe Asia/Atyrau"/>
```
Comment (mj1856) [inline common/supplemental/windowsZones.xml]: This edit is correct.

File: common/supplemental/windowsZones.xml [hunk 9]
```diff
@@ -736,9 +751,13 @@ For terms of use, see http://www.unicode.org/copyright.html
 			<mapZone other="Central Pacific Standard Time" territory="VU" type="Pacific/Efate"/>
 			<mapZone other="Central Pacific Standard Time" territory="ZZ" type="Etc/GMT-11"/>
 
-			<!-- (UTC+12:00) Anadyr, Petropavlovsk-Kamchatsky -->
-			<mapZone other="Russia Time Zone 11" territory="001" type="Asia/Kamchatka"/>
-			<mapZone other="Russia Time Zone 11" territory="RU" type="Asia/Kamchatka Asia/Anadyr"/>
+      <!-- (UTC+12:00) Anadyr, Petropavlovsk-Kamchatsky -->
+      <mapZone other="Russia Time Zone 11" territory="001" type="Asia/Anadyr"/>
+      <mapZone other="Russia Time Zone 11" territory="RU" type="Asia/Anadyr"/>
```
Comment (mj1856) [inline common/supplemental/windowsZones.xml]: This entry should not have been edited.

**Comment by yumaoka** [comment]
See #139

**Comment by ststeiger** [comment]
@yumaoka: This is fine with me. 

Just one thing about the obsolete timezones:
I use the edited file to translate timezone ids between Windows and Linux in my [fork of the EWS Managed API](https://github.com/ststeiger/ews-managed-api/blob/master/Microsoft.Exchange.WebServices.Data/TimeZoneData/TimeZoneTranslator.cs). 

Just because they're obsolete doesn't mean they cannot any no longer appear in the XML sent to/from Exchange-Web-Service. So any application relying on this file to translate timezones to and from windows will fail if an old system (be that Windows or Linux) still uses these timeones. 
I assume, this file has been created for a similar purpose. 

I wrote all the possible combinations to a database to see if all cases were covered (assuming windows 10), which they were not. 

So in my humble opinion, you should not remove obsolete mappings as long as they can still be in use, unless you intentionally want to break compatiblity in some edge-cases (cannot say if they actually occur or not). 

Wouldn't you better add an attribute isObsolete to the xml mappings as well ? 
Or use another tag name like <obsoleteMapZone, so thes can still be handled correctly under all circumstances.
 
Anyway, I can add the obsolete mappings manually, so my application doesn't break under these circumstances.

**Comment by mj1856** [comment]
@ststeiger - Since you're in C#, I recommend you use my [TimeZoneConverter](https://github.com/mj1856/TimeZoneConverter/) library instead.  It starts with the CLDR mappings, and also includes things [like a one-way mapping for the two deprecated zones](https://github.com/mj1856/TimeZoneConverter/blob/4a8159d9d7856f3f523c612c0198cbf00bf8cb27/src/TimeZoneConverter.DataBuilder/Program.cs#L86-L88).

**Pull request closed without merging**